### PR TITLE
Feature/alcs 90 part 2

### PR DESCRIPTION
**Pull request by mhuseinov**
* GRPC application service: submit application from portal to alcs
* GRPC document service: upload document from portal -> dellEcs, portal -> alcs
* Applicant info page in alcs
* Mutual TLS
* unit tests
* adjust deployment and unit tests for GRPC

### CODE REVIEW

File: portal-frontend/src/app/services/document/document.service.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DocumentService {
+  private serviceUrl = `${environment.apiUrl}/document`;
+  private httpClientNoAuth: HttpClient | null = null;
+
+  constructor(private httpClient: HttpClient, handler: HttpBackend) {
+    this.httpClientNoAuth = new HttpClient(handler);
```
Comment (dhaselhan) [inline portal-frontend/src/app/services/document/document.service.ts:15]: very clean way or working around the http interceptor!

File: services/apps/alcs/src/application/application-code/application-local-government/application-local-government.service.ts [hunk 2]
```diff
@@ -51,4 +51,15 @@ export class ApplicationLocalGovernmentService {
       },
     });
   }
+
+  async getByUuid(uuid: string) {
```
Comment (dhaselhan) [inline services/apps/alcs/src/application/application-code/application-local-government/application-local-government.service.ts:55]: Can you add a test for this new function?

File: services/apps/alcs/src/application/application-document/application-document.service.ts [hunk 3]
```diff
@@ -100,4 +102,25 @@ export class ApplicationDocumentService {
...
     return this.documentService.getDownloadUrl(document.document);
   }
+
+  async attachExternalDocuments(
+    fileNumber: string,
+    data: ApplicationDocumentCreateDto[],
+  ) {
+    const application = await this.applicationService.getOrFail(fileNumber);
+
+    const documentsToSave: ApplicationDocument[] = [];
```
Comment (dhaselhan) [inline services/apps/alcs/src/application/application-document/application-document.service.ts:112]: nitpick: I think it would be cleaner using a .map instead of creating a new array and using an of loop to push

File: services/apps/alcs/src/document/document.service.ts [hunk 4]
```diff
@@ -12,9 +12,10 @@ import { Repository } from 'typeorm';
 import { v4 } from 'uuid';
 import { CONFIG_TOKEN, IConfig } from '../common/config/config.module';
 import { User } from '../user/user.entity';
+import { CreateDocumentDto } from './document.dto';
 import { Document } from './document.entity';
 
-const DEFAULT_DB_TAGS = ['ORCS Classification: 85100-20'];
+export const DEFAULT_DB_TAGS = ['ORCS Classification: 85100-20'];
```
Comment (dhaselhan) [inline services/apps/alcs/src/document/document.service.ts:18]: Why did you add export here? I couldnt find it being imported anywhere.

File: services/apps/alcs/src/providers/typeorm/migrations/1670636483523-update_document_table.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    await queryRunner.query(
+      `ALTER TABLE "alcs"."document" DROP CONSTRAINT "FK_9b9254f412e8a5a07063209a08c"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "alcs"."document" ALTER COLUMN "uploaded_by_uuid" DROP NOT NULL`,
+    );
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
```
Comment (dhaselhan) [inline services/apps/alcs/src/providers/typeorm/migrations/1670636483523-update_document_table.ts:19]: Why is the down function dropping the same constraint that up also drops?

File: services/apps/portal/src/alcs/document-grpc/alcs-document.message.interface.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+export interface DocumentUploadResponseGrpc {
+  uploadUrl: string;
+  fileKey: string;
+}
+
+export class CreateDocumentRequestGrpc {
+  uploadedByUuid?: string;
+
+  mimeType: string;
```
Comment (dhaselhan) [inline services/apps/portal/src/alcs/document-grpc/alcs-document.message.interface.ts:16]: Does this also need @IsString()? Does that work here?

File: services/apps/portal/src/application/application-document/application-document.controller.ts [hunk 7]
```diff
@@ -110,4 +117,30 @@ export class ApplicationDocumentController {
     await this.applicationDocumentService.delete(document);
     return {};
   }
+
+  @Post('/application/:uuid/attachExternal')
+  async attachExternalDocument(
```
Comment (dhaselhan) [inline services/apps/portal/src/application/application-document/application-document.controller.ts:122]: Needs a unit test

File: services/apps/portal/src/application/application.service.ts [hunk 8]
```diff
@@ -69,6 +79,47 @@ export class ApplicationService {
...
+      );
+
+      //TODO set failed status here?
+
+      throw new BaseServiceException(
+        `Failed to submit application: ${fileNumber}`,
+      );
+    }
+
+    //TODO set submitted status here
```
Comment (dhaselhan) [inline services/apps/portal/src/application/application.service.ts:118]: Are you going to do another MR after this for submitted status?

Comment (mhuseinov) [reply services/apps/portal/src/application/application.service.ts:118]: nope, this is more placeholders of when we decide to add those statuses. At this point there is not clear requirement on what should happen and how it should look like after application is submitted

File: services/apps/alcs/src/providers/typeorm/migrations/1670636483523-update_document_table.ts (hunk 5 above)
Comment (mhuseinov) [reply services/apps/alcs/src/providers/typeorm/migrations/1670636483523-update_document_table.ts:19]: That is a good typeorm question 🙂 I'll check if I can regenerate it and if not I'll drop it

File: services/apps/portal/src/alcs/document-grpc/alcs-document.message.interface.ts (hunk 6 above)
Comment (mhuseinov) [reply services/apps/portal/src/alcs/document-grpc/alcs-document.message.interface.ts:16]: We do not need it on ALCS side since it is only used for GRPC. In portal controller dto and grpc_dto usually identical so it is possible to use one model with validator tags

File: services/apps/portal/src/application/application.service.ts (hunk 8 above)
Comment (dhaselhan) [reply services/apps/portal/src/application/application.service.ts:118]: Good point!

File: services/apps/portal/src/alcs/document-grpc/alcs-document.message.interface.ts (hunk 6 above)
Comment (dhaselhan) [reply services/apps/portal/src/alcs/document-grpc/alcs-document.message.interface.ts:16]: Okay thanks for explaining

**Review by dhaselhan — APPROVED** [review]
LGTM 👍
